# Day 13:30 daily statistics - Regressão Simples - Parte 4

$\color{red}{\text{Exercício 8}}$ - **Retirado do livro: Análise de Modelos de Regressão Linear com Aplicações, de Charnet, Freire, Charnet e Bonvino** - Dois testes de avaliação de personalidade foram aplicados a 20 indivíduos, obtendo-se os seguintes resultados (em pontos):

| Teste I(x) | Teste II (y) | Teste I(x) | Teste II (y) |
|:----------:|:------------:|:----------:|:------------:|
|     77     |      41      |     75     |      45      |
|     92     |      59      |     55     |      32      |
|     60     |      35      |     77     |      47      |
|     82     |      49      |     89     |      56      |
|     73     |      52      |     82     |      52      |
|     75     |      38      |     89     |      53      |
|     92     |      58      |     75     |      43      |
|     103    |      70      |     71     |      39      |
|     71     |      45      |     60     |      41      |
|     73     |      48      |     85     |      54      |

Construa intervalos de 90% de confiança para $\beta_0$, $\beta_1$ e $\sigma^2$.


In [1]:
#Entrando com os dados 
x=c(77,92,60,82,73,75,92,103,71,73,75,55,77,89,82,89,75,71,60,85)
y=c(41,59,35,49,52,38,58,70,45,48,45,32,47,56,52,53,43,39,41,54)
dados=data.frame(x,y)

#Ajuste do modelo
ajuste=lm(y~x,data=dados)
summary(ajuste)



Call:
lm(formula = y ~ x, data = dados)

Residuals:
    Min      1Q  Median      3Q     Max 
-7.8722 -2.0529  0.1791  1.3758  7.5406 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) -7.10571    5.83496  -1.218    0.239    
x            0.70637    0.07418   9.523 1.89e-08 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 3.85 on 18 degrees of freedom
Multiple R-squared:  0.8344,	Adjusted R-squared:  0.8252 
F-statistic: 90.68 on 1 and 18 DF,  p-value: 1.886e-08


In [2]:
#O intervalo de confiança para B0 e B1 são dados por:
confint(ajuste,level=0.9)

5 %         95 %     
(Intercept) -17.2239093 3.0124879
x             0.5777406 0.8350026

O intervalo de confiança para $\beta_0$ e $\beta_1$ ao nível de 90% de confiança é dado por: ($-17.2239093 \leq \beta_0\leq 3.0124879 $) e ($0.5777406 \leq \beta_1 \leq 0.8350026$).

$\color{red}{\text{Exercício 9}}$ - **Retirado do livro: Análise de Modelos de Regressão Linear com Aplicações, de Charnet, Freire, Charnet e Bonvino** - A tabela  a seguir apresenta a velocidade (em milhas por hora) e o consumo de combustível (em milhas por galão) em testes feitos com um certo tipo de carro.


| Velocidade | Consumo | Consumo | Consumo |
|:----------:|:-------:|:-------:|:-------:|
|     50     |   17,3  |   16,4  |         |
|     55     |   16,9  |   16,1  |    15   |
|     60     |   15,8  |   14,8  |   15,3  |
|     65     |   15,2  |   14,7  |         |
|     70     |   14,8  |   14,6  |   13,4  |
|     75     |   13,6  |   12,4  |         |
|     80     |   13,5  |   11,3  |   12,4  |
|     85     |   11,5  |         |         |
|     90     |   9,8   |   10,1  |   9,5   |
|     95     |   9,2   |   9,6   |    10   |
|     100    |   9,0   |   9,5   |   9,3   |

+ a) Ajuste um MRLS a estes dados;
+ b) Construa intervalos de 98% de confiança para os coeficientes do modelo ajustado;
+ c) Encontre um intervalo de 90% de confiança para o consumo médio de combustível, quando a velocidade for de 93 milhões por hora.

$\color{red}{\text{a)}}$ - Ajustar um MRLS a estes dados:

In [3]:
# Entrando com os dados 
x=seq(50,100,by=5)
vel=rep(x,c(2,3,3,2,3,2,3,1,3,3,3))
con=c(17.3,16.4,16.9,16.1,15,15.8,14.8,15.3,15.2,14.7,14.8,14.6,13.4,13.6,12.4,13.5,11.3,12.4,11.5,9.8,10.1,9.5,9.2,9.6,10,9,9.5,9.3)
dados9=data.frame(vel,con)

#Realizando o ajuste no modelo
ajuste9=lm(con~vel,data=dados9)
summary(ajuste9)


Call:
lm(formula = con ~ vel, data = dados9)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.21696 -0.55153  0.08445  0.43608  1.32977 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 25.119780   0.592740   42.38   <2e-16 ***
vel         -0.161869   0.007673  -21.09   <2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.6564 on 26 degrees of freedom
Multiple R-squared:  0.9448,	Adjusted R-squared:  0.9427 
F-statistic:   445 on 1 and 26 DF,  p-value: < 2.2e-16


O modelo estimado apresenta $\beta_0=25.119780$ e $\beta_1=-0.161869$. Portanto o modelo estimado é dado por $\hat{y}=25.119780-0.161869X$.

$\color{red}{\text{b)}}$ - Coeficientes de confiança de 98% para os parâmetros estimados

In [4]:
confint(ajuste9,level=0.98)

1 %        99 %      
(Intercept) 23.6505979 26.5889628
vel         -0.1808889 -0.1428499

O intervalo de confiança para $\beta_0$ e $\beta_1$ são respectivamente: ($23.6505979 \leq \beta_0 \leq 26.5889628$) e $(-0.1808889 \leq \beta_1 \leq -0.1428499)$.

$\color{red}{\text{c)}}$ - Intervalo de 90% de confiança para o consumo médio de combustível, quando a velocidade for de 93 milhas por hora

In [5]:
velocidade=data.frame(vel=93)
predict(ajuste9, velocidade,interval="confidence",level=0.90)

fit      lwr      upr     
1 10.06592 9.754464 10.37738

O consumo médio de combustível quando a velocidade for de 93 milhas por hora é 10.06 (mi/gal), variando entre (9,754464;10.37738) ao nível de 90% de confiança.

$\color{red}{\text{Exercício 10}}$ - **Retirado do livro: Análise de Modelos de Regressão Linear com Aplicações, de Charnet, Freire, Charnet e Bonvino** - As tabelas abaixo fazem parte da saída de uma análise de regressão realizada pelo Minitab

| Predictor |   Coef  |  Stdev | t-ratio |   p   |
|:---------:|:-------:|:------:|:-------:|:-----:|
|  Constant |  83.074 |  6.593 |  12.60  | 0.000 |
|     x     | -1.1848 | 0.1258 |  -9.42  | 0.000 |

e

|   SOURCE   | DF |   SS   |   MS   |   F   |   P   |
|:----------:|:--:|:------:|:------:|:-----:|:-----:|
| Regression |  1 | 1021.1 | 1021.1 | 88.68 | 0.000 |
|    Error   |  7 |  80.6  |  11.5  |       |       |
|    Total   |  8 | 1101.6 |        |       |       |


+ a) Escreva a equação da reta ajustada
+ b) Encontre um intervalo de confiança de 95%, para o coeficiente angular da reta;
+ c) Verifique se o modelo contribui para explicar a variável resposta;
+ d) Encontre SSE
+ e) O modelo é adequado? Justifique a sua resposta.

$\color{red}{\text{a)}}$ A equação ajustada é dada por: $\hat{y}=83.074-1.1848X$

$\color{red}{\text{b)}}$ O intervalo de confiança de 95% para o coeficiente angular estimado é dado por: $\hat{\beta_1}\pm t_{((1-\alpha/2);(n-2))}S_e\sqrt{\frac{1}{\sum(x_i-\overline{x})^2}}$ 

Temos: $$SQ_{reg}=\hat{\beta_1}^2\cdot \sum (x_i-\overline{x})^2$$
$$1021,1 = (-1,1848)^2\cdot \sum(x_i-\overline{x})^2$$
$$\sum(x_i-\overline{x})^2=\frac{1021,1}{(-1,1848)^2}$$
$$\sum(x_i-\overline{x})^2=727.4082$$

Logo, com os resultados acima, o valor de $\sum(x_i-\overline{x})^2=727.4082$. Com isso o intervalo de confiança para $\beta_1$ ao nível de 95% de confiança é dado por:

$$IC[\beta_1;0,95]=-1.1848 \pm (2,4469)\cdot \sqrt{11,5}\cdot \sqrt{\frac{1}{727,4082}}$$
$$IC[\beta_1;0,95]=[-1,492464 ; -0,8771365]$$

$\color{red}{\text{c)}}$ -  O coeficiente de determinação para o modelo estimado é dado por:

$$R^2=\frac{SQ_{reg}}{SQ_{total}}$$
$$R^2=\frac{1021.1}{1101.6}$$
$$R^2=0,9269$$

Portanto o modelo estimado explica em 92,69% a variação que é inerente ao processo.

$\color{red}{\text{d)}}$ -  Determine o SSE.

De acordo com a tabela da Anova acima o SSE é dado por 80,6

$\color{red}{\text{e)}}$ - O modelo poderá ser considerado adequado?

Os parâmetros estimados para o modelo ($\beta_0$) e ($\beta_1$) apresentaram os p-valores menores do que 0,00, indicando que ambos podem ser considerados significativos para o modelo. O modelo estimado apresenta um coeficiente de determinação de 92,69% e uma variância residual de 11.5 em torno da reta média estimada. Ao realizar a ANOVA para a regressão o p-valor associado ao modelo foi igual a zero, nos levando a rejeitarmos a hipótese de não significancia do modelo, ou seja, existem evidencias amostrais ao nível de 95% de confiança que nos garantem que o modelo é significativo.